packages/build-info/src/project.ts

```diff
@@ -308,15 +319,24 @@ export class Project {
                 // noop framework not found
               }
             } else if (this.workspace) {
-              const result = await this.detectFrameworksInPath(this.fs.join(this.workspace.rootDir, pkg))
-              this.frameworks.set(pkg, result)
+              const results = await this.detectFrameworksInPath(this.fs.join(this.workspace.rootDir, pkg))
+              // if we detect a framework (with the same detection result in a path, instead of the root we need to remove it in the root)
+              for (const result of results) {
+                for (let i = 0, max = rootFrameworks.length; i < max; i++) {
+                  if (isEqual(result.detected, rootFrameworks[i].detected)) {
+                    rootFrameworkMutex.runExclusive(async () => {
```

Comment:
can you briefly explain why mutex is needed here (and only here)?

Reply:
also - shouldn't this exclusive run be wider? index of item to remove is figured before entering exclusive mode and if mutex is here at all, presumably multiple concurrent runs can execute this code, so index figured out before acquiring mutex might no longer be correct because since figured out `i` something else might have mutated things?